services/m3dbnode/server/server.go

```diff
@@ -226,38 +229,76 @@ func Run(runOpts RunOptions) {
-				SetLogger(logger).
-				SetMetricsScope(scope))
-	configSvcClient, err := etcdclient.NewConfigServiceClient(configSvcClientOpts)
-	if err != nil {
-		logger.Fatalf("could not create m3cluster client: %v", err)
-	}
+	var (
+		topoInit topology.Initializer
+		kv       m3clusterkv.Store
+	)
+	switch {
+	case cfg.ConfigService != nil:
+		logger.Info("creating dynamic config service client with m3cluster")
+		configSvcClientOpts := cfg.ConfigService.NewOptions().
+			SetInstrumentOptions(
+				instrument.NewOptions().
+					SetLogger(logger).
+					SetMetricsScope(scope))
+		configSvcClient, err := etcdclient.NewConfigServiceClient(configSvcClientOpts)
+		if err != nil {
+			logger.Fatalf("could not create m3cluster client: %v", err)
+		}
 
-	dynamicOpts := namespace.NewDynamicOptions().
-		SetInstrumentOptions(iopts).
-		SetConfigServiceClient(configSvcClient).
-		SetNamespaceRegistryKey(kvconfig.NamespacesKey)
-	nsInit := namespace.NewDynamicInitializer(dynamicOpts)
+		dynamicOpts := namespace.NewDynamicOptions().
+			SetInstrumentOptions(iopts).
+			SetConfigServiceClient(configSvcClient).
+			SetNamespaceRegistryKey(kvconfig.NamespacesKey)
+		nsInit := namespace.NewDynamicInitializer(dynamicOpts)
 
-	opts = opts.SetNamespaceInitializer(nsInit)
+		opts = opts.SetNamespaceInitializer(nsInit)
 
-	serviceID := services.NewServiceID().
-		SetName(cfg.ConfigService.Service).
-		SetEnvironment(cfg.ConfigService.Env).
-		SetZone(cfg.ConfigService.Zone)
+		serviceID := services.NewServiceID().
+			SetName(cfg.ConfigService.Service).
+			SetEnvironment(cfg.ConfigService.Env).
+			SetZone(cfg.ConfigService.Zone)
 
-	topoOpts := topology.NewDynamicOptions().
-		SetConfigServiceClient(configSvcClient).
-		SetServiceID(serviceID).
-		SetQueryOptions(services.NewQueryOptions().SetIncludeUnhealthy(true)).
-		SetInstrumentOptions(opts.InstrumentOptions()).
-		SetHashGen(sharding.NewHashGenWithSeed(cfg.HashingConfiguration.Seed))
+		topoOpts := topology.NewDynamicOptions().
+			SetConfigServiceClient(configSvcClient).
+			SetServiceID(serviceID).
+			SetQueryOptions(services.NewQueryOptions().SetIncludeUnhealthy(true)).
+			SetInstrumentOptions(opts.InstrumentOptions()).
+			SetHashGen(sharding.NewHashGenWithSeed(cfg.HashingConfiguration.Seed))
+
+		topoInit = topology.NewDynamicInitializer(topoOpts)
+
+		kv, err = configSvcClient.KV()
+		if err != nil {
+			logger.Fatalf("could not create KV client, %v", err)
+		}
+
+	case cfg.StaticConfig != nil && cfg.StaticConfig.TopologyConfig != nil:
+		logger.Info("creating static config service client with m3cluster")
+
+		shardSet, hostShardSets, err := createStaticShardSet(cfg.StaticConfig.TopologyConfig.Shards, cfg.ListenAddress)
+		if err != nil {
+			logger.Fatalf("unable to create shard set for static config: %v", err)
+		}
+		staticOptions := topology.NewStaticOptions().
+			SetReplicas(1).
+			SetHostShardSets(hostShardSets).
+			SetShardSet(shardSet)
+
+		md, err := createDefaultMetaData("metrics")
```

Comment:
I think this should be in config, in fact it should probably be able to have multiple namespaces as well.  

We can expand `StaticConfiguration` in the config package to be something like: